components/ingest-service/pipeline/publisher/config_mgmt_publisher.go

```diff
@@ -0,0 +1,92 @@
+package publisher
+
+import (
+	"context"
+	"time"
+
+	"github.com/chef/automate/api/interservice/cfgmgmt/request"
+	cfgmgmt "github.com/chef/automate/api/interservice/cfgmgmt/service"
+	"github.com/chef/automate/components/ingest-service/pipeline/message"
+	log "github.com/sirupsen/logrus"
+)
+
+func BuildConfigMgmtPublisher(client cfgmgmt.CfgMgmtClient, numPublishers int) message.ChefActionPipe {
+	return func(in <-chan message.ChefAction) <-chan message.ChefAction {
+		if numPublishers <= 0 || client == nil {
+			log.Info("Direct publish to config management service is disabled")
+			return actionsNoop(in)
+		}
+
+		out := make(chan message.ChefAction, 100)
+		log.Infof("Starting config-management-service publisher")
+		for i := 0; i < numPublishers; i++ {
+			go configMgmtPublisher(in, client, out)
+		}
+		return out
+	}
+}
+
+func configMgmtPublisher(in <-chan message.ChefAction, client cfgmgmt.CfgMgmtClient, out chan<- message.ChefAction) {
+	for msg := range in {
+		cmpHandleMessage(client, msg, out)
+
+	}
+	close(out)
+}
+
+func cmpHandleMessage(client cfgmgmt.CfgMgmtClient, msg message.ChefAction, out chan<- message.ChefAction) {
+	if err := msg.Ctx.Err(); err != nil {
+		msg.FinishProcessing(err)
+		return
+	}
+
+	defer message.PropagateChefAction(out, &msg)
+
+	if !isPolicyUpdateAction(msg) {
+		return
+	}
+
+	start := time.Now()
+
+	req := &request.PolicyUpdateAction{
+		PolicyName:         msg.InternalChefAction.EntityName,
+		PolicyGroup:        msg.InternalChefAction.ParentName,
+		PolicyRevisionId:   msg.InternalChefAction.RevisionId,
+		ChefServerFqdn:     msg.InternalChefAction.ServiceHostname,
+		ChefServerOrgname:  msg.InternalChefAction.OrganizationName,
+		ChefServerUsername: msg.InternalChefAction.RequestorName,
+		PolicyfileContent:  msg.InternalChefAction.Data,
```

Comment:
I will just parse out all the cookbooks from the "Data" field and add a new field to the PolicyUpdateAction.